### ISSUE TITLE
Can't submit metaSPAdes_assemble script when using the my_data_set.yaml file?

### ISSUE THREAD

**Issue by B-1991-ing** [ablab/spades#982]
### Is your feature request related to a problem? Please describe. For generic questions use Q&A section in the Discussions forum above.

Dear metaSPAdes support team,

I am now assembling 9 paired-end samples using command line below. But, as my sample id is inconsistent, eg., 25,27, 29, 31-36. So, I use the **.yaml** file as the input of --dataset. But, if I write the absolute path of paired-end .fq.gz files, then I can't use the array id in my PBS job setting. For example, I can specify setting like, **PBS -t 25,27,29,31-36%3** or ${PBS_ARRAYID}.
Could you give some suggestions on my question?

**Command line**
/services/tools/SPAdes/3.15.3/bin/spades.py --meta -k 21,33,55,77 --dataset /PATH_TO_YAML_FILE/my_data_set.yaml -t 40 -m 1500 -o /PATH_TO_OUPUT/spades_output_25_36

**--dataset - my_data_set.yaml.txt**
[my_data_set copy.yaml.txt](https://github.com/ablab/spades/files/8837844/my_data_set.copy.yaml.txt)

Best,

Bing

**Comment by asl** [comment]
Sorry, I do not follow what the problem is. Maybe you'd just generate YAML from your PBS script?

**Comment by B-1991-ing** [comment]
Thank you very much for replying me. 

My question is that I can't directly run the command line with **${PBS_ARRAYID}** as below due to I am using the **"my_data_set.yaml"** dataset file.

The command line with **${PBS_ARRAYID}**
/services/tools/SPAdes/3.15.3/bin/spades.py --meta -k 21,33,55,77 --pe${PBS_ARRAYID}-1 Sample${PBS_ARRAYID}_TM_paired_1.fastq --pe${PBS_ARRAYID}-2 Sample${PBS_ARRAYID}_TM_paired_2.fastq -t 32 -o ./Current_pipeline/metaSPAdes_assemble/Sample${PBS_ARRAYID}

The command line with **"my_data_set.yaml"** dataset file, which I am running now.
/services/tools/SPAdes/3.15.3/bin/spades.py --meta -k 21,33,55,77 --dataset /home/projects/env_10000/people/binson/MTB/assemble/metaSPAdes_assemble/my_data_set.yaml -t 40 -o /home/projects/env_10000/people/binson/MTB/assemble/metaSPAdes_assemble/spades_output_25_36

Best,

Bing

**Comment by asl** [comment]
Well, exactly as I said: you'd need to generate YAML file from your PBS script. And use it for SPAdes run.

Hope it helps

**Comment by B-1991-ing** [comment]
Thank you very much for your quick response.

I need to generate one YAML file for each of my paired-end file like below?
/services/tools/SPAdes/3.15.3/bin/spades.py --meta -k 21,33,55,77 --dataset /home/projects/env_10000/people/binson/MTB/assemble/metaSPAdes_assemble/my_data_set_**${PBS_ARRAYID}**.yaml -t 40 -o /home/projects/env_10000/people/binson/MTB/assemble/metaSPAdes_assemble/spades_output_25_36

When I submit my PBS shell script on HPC, can I submit with 6 inconsistent job ids - "PBS -t 25,27,29,31-36%3"? Then, I can submit 3 jobs at the same time?

**Comment by asl** [comment]
> I need to generate one YAML file for each of my paired-end file like below?

No. You need to generate *the contents* of this file as necessary

> When I submit my PBS shell script on HPC, can I submit with 6 inconsistent job ids - "PBS -t 25,27,29,31-36%3"?

No idea. You may want to ask your system administrator about this.
